cognite/powerops/client/shop/api/dayahead_trigger_api.py

```diff
@@ -15,58 +22,139 @@ def __init__(self, client: CogniteClient, data_set: int, cogshop_version: str =
+                },
+            )
+        )
+        events_and_relationships["relationships"].append(
+            Relationship(
+                external_id=f"{workflow_event_external_id}.{total_event_external_id}",
+                source_external_id=workflow_event_external_id,
+                source_type="event",
+                target_external_id=total_event_external_id,
+                target_type="event",
+                data_set_id=self._data_set_api,
+                labels=[TotalFunctionEvent.relationship_label_to_trigger_event],
+            )
+        )
+
+        return events_and_relationships
+
+    def _create_trigger_event(self, workflow: DayaheadTrigger, start_time: datetime, bid_date: str) -> Event:
+        """
+        Create a workflow trigger event and link the shop runs to this event.
+        (needs method, price scenarios and plant for calculating total bid matrix per plant later on)
+        Returns:
+            Event used to track the tasks linked to the workflow
+        """
+        return DayaheadTriggerEvent.as_cdf_event(
+            data_set=self._data_set_api,
+            start_time=start_time,
+            bid_date=bid_date,
+            price_area=workflow.price_area,
+            price_scenarios=workflow.price_scenarios,
+            method=workflow.method,
+        )
+
+    def _create_and_wire_workflow_events(
+        self, workflow: DayaheadTrigger, shop_runs_as_cdf_events: list[Event], start_time: datetime, bid_date: str
+    ) -> dict[str, Event | list[Event]]:
+        workflow_event = self._create_trigger_event(workflow, start_time, bid_date)
+        events_and_relationships = self._create_function_events(
+            workflow=workflow, workflow_event_external_id=workflow_event.external_id, bid_date=bid_date
+        )
+        self._client.events.create([workflow_event, *events_and_relationships["events"]])
+
+        shop_relationships = [
+            Relationship(
+                external_id=f"{workflow_event.external_id}.{shop_event.external_id}",
+                source_external_id=workflow_event.external_id,
+                source_type="event",
+                target_external_id=shop_event.external_id,
+                target_type="event",
+                data_set_id=self._data_set_api,
+                labels=[DayaheadTriggerEvent.relationship_label_to_shop_run_events],
+            )
+            for shop_event in shop_runs_as_cdf_events
+        ]
+        self._client.relationships.create([*events_and_relationships["relationships"], *shop_relationships])
+
+        return {
+            "workflow_trigger_event": workflow_event,
+            "bid_matrix_function_events": events_and_relationships["events"],
+            "shop_run_events": shop_runs_as_cdf_events,
+        }
 
-    def trigger_workflow(self, workflow: DayaheadTrigger) -> dict:
+    def trigger_workflow(self, workflow: DayaheadTrigger) -> dict[str, Event | list[Event]]:
```

Comment:
I am in doubt about the return type here. How is the user expected to use this? Should we introduce some dataclass as I see this will always have three keys, `workflow_trigger_event`, `bid_matrix_function_events`, and `shop_run_events`. Suggest calling this class something like  `DayaheadTriggerResults`.